strawberry/flask/views.py

```diff
@@ -92,3 +92,55 @@ def dispatch_request(self) -> Response:
         response.set_data(json.dumps(response_data))
 
         return response
+
+
+class AsyncGraphQLView(GraphQLView):
+    async def dispatch_request(self):
+        method = request.method
+        content_type = request.content_type or ""
+
+        if "application/json" in content_type:
+            data: dict = request.json  # type:ignore[assignment]
+        elif content_type.startswith("multipart/form-data"):
+            operations = json.loads(request.form.get("operations", "{}"))
+            files_map = json.loads(request.form.get("map", "{}"))
+
+            data = replace_placeholders_with_files(operations, files_map, request.files)
+        elif method == "GET" and request.args:
+            data = parse_query_params(request.args.to_dict())
+        elif method == "GET" and should_render_graphiql(self.graphiql, request):
+            template = render_graphiql_page()
+
+            return self.render_template(template=template)
+        else:
+            return Response("Unsupported Media Type", 415)
+
+        try:
+            request_data = parse_request_data(data)
+        except MissingQueryError:
+            return Response("No valid query was provided for the request", 400)
+
+        response = Response(status=200, content_type="application/json")
+        context = self.get_context(response)
+
+        allowed_operation_types = OperationType.from_http(method)
+
+        if not self.allow_queries_via_get and method == "GET":
+            allowed_operation_types = allowed_operation_types - {OperationType.QUERY}
```

Comment:
@patrick91 I'm super conscious of the fact that this is an exact copy of the logic in the existing view. 

As a potential solution to this we could overload the dispatch_request to be optionally async. I have found [this](https://stackoverflow.com/questions/39778901/optional-synchronous-interface-to-asynchronous-functions) as a guide to be a potential solution for this, however it has some noticeable defects. 

I'm happy to restructure this code to whatever will be desirable to you.

Reply:
no worries at all! we'd like to refactor __all__ the views to reduce the code duplication at some point. Hopefully we find a solution for this, but even if we don't we won't have as much duplicated code as we have now :)

I've done some precursory work here (just for the tests), if you're curious: https://github.com/strawberry-graphql/strawberry/pull/1840